torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -1221,7 +1267,11 @@ void AliasDb::giveFreshAlias(
   auto new_elem = memoryDAGBuilder_->makeFreshValue(value);
   elementMap_[value] = new_elem;
   if (add_wildcard_to_contained_elems) {
-    addContainedTypesToFreshElement(new_elem, *maybe_mut_type);
+    if ((*maybe_mut_types).size() > 1) {
+      pointUnionTypeElementToAllContainedTypes(new_elem, *maybe_mut_types);
```

Comment:
nit: I would add an assertion that the type is a Union type here. I dont know if we have Optional[Union] right now (eek) so maybe this is a todo after we remove optionals

Reply:
It's not possible to have `Optional[Union]`! I changed that in the first PR to always be `Union[None, ...]`